src/apps/applets/crud/applets.py

```diff
@@ -119,20 +129,17 @@ async def get_applets_by_roles(
         result: Result = await self._execute(query)
         return result.scalars().all()
 
-    async def get_applet_by_role(
+    async def get_applet_by_roles(
         self, user_id: int, applet_id: int, roles: list[str]
-    ) -> AppletSchema:
+    ) -> AppletSchema | None:
         query = select(AppletSchema)
         query = query.join_from(UserAppletAccessSchema, AppletSchema)
         query = query.where(AppletSchema.id == applet_id)
         query = query.where(UserAppletAccessSchema.user_id == user_id)
         query = query.where(UserAppletAccessSchema.role.in_(roles))
         query = query.limit(1)
         result: Result = await self._execute(query)
-        try:
-            return result.scalars().one()
-        except Exception as e:
-            raise e
+        return result.scalars().first()
```

Comment:
Don't you think it is better to raise NotFoundError instead of returning the None object?

Reply:
First time I was agree with it, but in some cases I needed to reuse the same method and raise another error. Moreover, if role is not assigned to this user, so it means no not found, it means access denied.